TESTS/mbed_hal/ticker_api_tests/main.cpp

```diff
@@ -0,0 +1,520 @@
+ *
+ * For the boards with ticker clock freq less than or equal to 250 KHz we will try to prove that ticker is incremented by one
+ * straightforward by reading ticker count value in the loop in order to detect a single ticker value update (hopefully by one).
+ * For faster tickers we need to prove this indirectly using additional count_ticks() function which returns number of
+ * ticks needed to perform N cycles of the empty while loop. For the same number of cycles function result should be the same with
+ * accuracy +/- 1 tick. After the first test we will call count_ticks again with number of cycles equal N, N+1, N+2, ...
+ * until we get other ticks result.
+ *
+ * Given ticker is available.
+ * When ticker is initialized.
+ * Then ticker counter is incremented by one.
+ */
+void ticker_increment_test(void)
+{
+    /* Init the ticker. */
+    intf->init();
+
+    /* Get ticker info. */
+    const ticker_info_t* p_ticker_info = intf->get_info();
+
+    /* Perform test based on ticker speed. */
+    if (p_ticker_info->frequency <= 250000) {    // low frequency tickers
+        const uint32_t base_tick_count = intf->read();
+        uint32_t next_tick_count = base_tick_count;
+
+        while (next_tick_count == base_tick_count) {
+            next_tick_count = intf->read();
+        }
+
+        TEST_ASSERT_UINT32_WITHIN(1, next_tick_count, base_tick_count);
+    } else {  // high frequency tickers
+        const uint32_t base_tick_count = count_ticks(NUM_OF_CYCLES, 1);
+        uint32_t next_tick_count = base_tick_count;
+        uint32_t inc_val = 0;
+
+        while (next_tick_count == base_tick_count) {
+            next_tick_count = count_ticks(NUM_OF_CYCLES + inc_val, 1);
+            inc_val++;
+        }
+
+        /* Since we are here we know that next_tick_count != base_tick_count.
+         * The accuracy of our measurement method is +/- 1 tick, so it is possible that
+         * next_tick_count == base_tick_count - 1. This is also valid result.
+         */
+        TEST_ASSERT_UINT32_WITHIN(1, next_tick_count, base_tick_count);
+
+    }
+}
+
+/* Test that common ticker functions complete with the required amount of time.
+ *
+ * Given ticker is available.
+ * When ticker_read, ticker_clear_interrupt, ticker_set_interrupt, ticker_fire_interrupt or ticker_disable_interrupt function is called.
+ * Then its execution is not longer than 20 us.
+ */
+void ticker_speed_test(void)
+{
+    Timer timer;
+    int counter = NUM_OF_CALLS;
+
+    /* Init the ticker. */
+    intf->init();
+
+    /* ---- Test ticker_read function. ---- */
+    timer.reset();
+    timer.start();
+    while (counter--) {
+        intf->read();
+    }
+    timer.stop();
+
+    /* Check result. */
+    TEST_ASSERT(timer.read_us() < (NUM_OF_CALLS * (MAX_FUNC_EXEC_TIME_US + DELTA_FUNC_EXEC_TIME_US)));
+
+    /* ---- Test ticker_clear_interrupt function. ---- */
+    timer.reset();
+    timer.start();
+    while (counter--) {
```

Comment:
I guess you forgot to reset the counter to `NUM_OF_CALLS`.

Reply:
Fixed.